hack/verify-api-reference-docs.sh

```diff
@@ -25,12 +25,18 @@ source "${KUBE_ROOT}/hack/lib/init.sh"
 
 kube::golang::setup_env
 
+
 API_REFERENCE_DOCS_ROOT="${KUBE_ROOT}/docs/api-reference"
+# Use REPO_DIR if provided so we can set it to the host-resolvable path
+# to the repo root if we are running this script from a container with
+# docker mounted in as a volume.
+REPO_DIR=${REPO_DIR:-"${KUBE_ROOT}"}
+ABS_OUTPUT_DIR="${REPO_DIR}/_tmp/api-reference"
 TMP_OUTPUT_DIR="${KUBE_ROOT}/_tmp/api-reference"
 TMP_ROOT="${KUBE_ROOT}/_tmp"
 
 # Generate API reference docs in tmp.
-"./hack/update-api-reference-docs.sh" "${TMP_OUTPUT_DIR}"
+"./hack/update-api-reference-docs.sh" "${ABS_OUTPUT_DIR}"
```

Comment:
The name was a bit misleading. They're actually both absolute paths, but we need to pass in a host-resolvable path to update-api-reference-docs.sh for this script to run in a container. Renamed to HOST_OUTPUT_DIR and added comment explaining why it's necessary.

There is a [comment](https://github.com/kubernetes/kubernetes/pull/16287/files#diff-b38b0e262cb33e5d5cf7c6269785a970R30) to this effect, but gotest-dockerized does a `docker run` with docker itself mounted into the container as a volume so that tests/scripts (running in a container) can use also launch containers. This particular script does a `docker run -v` and mounts a path under the repo directory. In order to do that, it has to use the host-resolvable repo path for the docker mount path. However the diff runs in the kubekins-test, so it should use the repo path resolvable from the container. If it helps, this is what's going on:

[host]
path to repo: `/home/me/kubernetes`
cmd:
```
    cd ~ kubernetes
    hack/jenkins/gotest-dockerized.sh 
    ...
    # launches kubekins-test container
```

[ inside kubekins-test container]
path to repo: `/go/src/k8s.io/kubernetes`
Note that the above path is mapped to `/home/me/kubernetes` on the host.
cmd:
```
  cd /go/src/k8s.io/kubernetes
  ...
  hack/verify-api-reference-docs.sh
  ...
  # launches gendocs container
```

[ inside gendocs container ]
path to output dir: `/output`
Note that `/output` is mapped to `/home/me/kubernetes/docs/api-reference` on the host, and `/go/src/k8s.io/kubernetes/docs/api-reference` inside kubekins-test.

   cmd:
```
   build/gen-swagger-docs.sh
```

Hope that makes sense. The inception is a bit tricky, so I've tried to leave comments explaining it.